src/api/Folder.js

```diff
@@ -207,19 +206,19 @@ class Folder extends Item {
      * @return {Promise}
      */
     folderRequest(
-        { fields, noPagination }: FetchOptions = {},
+        { fields, noPagination, shouldFetchThumbnails }: FetchOptions = {},
         successHandler?: Function = this.folderSuccessHandler,
     ): Promise<any> {
         if (this.isDestroyed()) {
             return Promise.reject();
         }
 
         const requestFields = fields || FOLDER_FIELDS_TO_FETCH;
-
-        this.errorCode = ERROR_CODE_FETCH_FOLDER;
-        if (SHOULD_FETCH_THUMBNAILS) {
+        if (shouldFetchThumbnails) {
```

Comment:
I wonder how necessary this approach is vs just adding the `FIELD_REPRESENTATIONS` to the `FOLDER_FIELDS_TO_FETCH`. Could consider doing some comparison between calls with and without representations to see how measurable this performance impact is

Reply:
I ran 10 trials with and without the reps field.  Average time without the reps field was `2.00s`, average time with the reps field was `2.52s`.